File: algorithms/1_easy/M/00896_monotonic_array/solution.c
bool isMonotonic(const int* nums, const int len) {
    if (len <= 1) {
        return true;
    }

    if (nums == NULL) {
        fprintf(stderr, "invalid input\n");
        exit(1);
    }

    int state = 0;

    for (int i = 1; i < len; i += 1) {
        if (nums[i] > nums[i - 1]) {
            if (state == -1) {
                return false;
            }
            state = 1;
        } else if (nums[i] < nums[i - 1]) {
            if (state == 1) {
                return false;
            }
            state = -1;
        }
    }

    return true;
}

File: algorithms/1_easy/M/00896_monotonic_array/solution.rs
impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
        if nums.len() <= 1 {
            return true;
        }

        let len = nums.len();
        let mut state = 0_i32;

        for i in 1..len {
            if nums[i] > nums[i - 1] {
                if state == -1 { 
                    return false;
                }
                state = 1;
            } else if nums[i] < nums[i - 1] {
                if state == 1 {
                    return false;
                }
                state = -1;
            }
        }

        true
    }
}
